packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -4617,10 +4623,21 @@ class StatefulElement extends ComponentElement {
     return super.dependOnInheritedElement(ancestor as InheritedElement, aspect: aspect);
   }
 
+  /// This controls whether we should call [State.didChangeDependencies] from
+  /// the start of [build], to avoid calls when the [State] will not get built.
+  /// This can happen when the widget has dropped out of the tree, but depends
+  /// on an [InheritedWidget] that is still in the tree.
```

Comment:
How would this possible? if an element dropped out the tree, wouldn't it call deactivate to remove the subscription?

Reply:
oh i see, it happens before the subtree is built